minima/database.nim

```diff
@@ -75,20 +92,21 @@ proc open*(dir: string): Result[Database, DatabaseError] =
     if fileExists(path):
         mode = fmReadWriteExisting
 
-    var log: File
+    var f: File
     try:
-        log = open(path, mode)
+        f = open(path, mode)
     except:
         return err(TreeFileCreationFailed)
 
     var db = Database(
         dir: dir,
-        log: log,
+        log: StandardLog.init(f),
```

Comment:
It seems that you can remove the code duplication between the two `open` functions by creating a helper that takes an already initialized `Log` parameter.